### PR TITLE
[WIP] Add ServiceInsight and ServicePulse

### LINKED ISSUE #89: Add ServiceInsight and ServicePulse
In the last delivery of the workshop at NDC Oslo 2017, I instructed attendees to look at ServiceInsight as a part of the exercises that they were doing, as well as adding the Saga Audit plugin for the saga exercises.

For the existing NServiceBus users/customers who attended, experiencing Pulse and Insight first-hand was very significant. I heard them saying that there was no way that they'd go back to working without them, almost regardless of how much they cost.

In short, I propose that we incorporate more instruction into the workshop around how to install, configure, and use Pulse and Insight in the workshop.

I'd also suggest that this be prioritized so that it can be done in time for the next delivery of the workshop which, I believe, is Jimmy doing it in NDC Sydney.

### PR DESCRIPTION
**Pull request by adamralph**
Fixes #89

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,6 +103,39 @@ The solutions contain single page applications (SPAs) and use `IIS Express`. To
...
+In the settings for the new instance, there are two choices you must make:
+
+- "Transport": choose "MSMQ".
+- "Audit forwarding": choose "Off".
+
+All other settings should be left with their default values.
+
+Click "Add".
+
+After the instance has been added, make a note of the name of the instance. This will usually be "Particular.ServiceControl".
```
Comment (mauroservienti) [inline README.md]: Is this needed? That's the default value, meaning that installing the heartbeat plugin is enough and no further configuration is needed

Comment (adamralph) [inline README.md]: Is it not needed when configuring the saga audit plug in?

Comment (mauroservienti) [inline README.md]: That specifically I don't remember, but it should not

Comment (adamralph) [inline README.md]: From [the docs](https://docs.particular.net/servicecontrol/plugins/saga-audit#configuration-servicecontrol-queue):

> The queue name is based on the ServiceControl instance name...

which means you have to configure it like so:

```C#
var endpointConfiguration = new EndpointConfiguration("myendpoint");
endpointConfiguration.SagaPlugin("Particular.ServiceControl");
```

Comment (mauroservienti) [inline README.md]: Yup. Had a look at the source code, and there isn't anymore a default value

**Comment by dvdstelt** [comment]
Hmmm, wait a minute. I approved this, but the advanced exercises aren't there yet, are they!? :-)
I suggest we add the documentation, but don't create any "before" or "after" folders. If anything goes wrong, this is an excellent chance to read documentation and get hands-on experience. And they are 'advanced' after all, right?

**Comment by adamralph** [comment]
@dvdstelt 

> Hmmm, wait a minute. I approved this, but the advanced exercises aren't there yet, are they!? :-)

Correct, that's why it's marked WIP.

> I suggest we add the documentation, but don't create any "before" or "after" folders.

The idea (as I understand it) is to use the existing solutions (1 to 4) and only to add some instructions for advanced exercises that will involve using SP and SI. I don't think we should alter the solutions at all if we can avoid it. You could argue that we should alter the "after" solution to reflect the state after the advanced exercises have been completed, but I would argue that this isn't required. Those who are moving on to the advanced exercises can be left a little more "on their own", without a reference "after" solution to look at for clues.

/cc @mauroservienti

**Comment by dvdstelt** [comment]
Please review advanced exercise. I added @tmasternak and @SzymonPobiega because they might want to know about it as well, but it is very unlikely this will come up in a 1 day workshop! But you never know, some devs are so extremely fast! ;-)

**Comment by dvdstelt** [comment]
Aaargh, some files are included that certainly should not have been. Let me try to fix.

For the reviewers, it only concerns the new .md file! :)

**Comment by dvdstelt** [comment]
Hmmm, I thought I had already replied to this thread.

- Module 03 already has advanced exercises.
- Module 04 doesn't need them, as this is the advanced exercise
- There is no solution to maintain, They should pick this up after they finish 04. Or worst case scenario they pick the `after` solution of module 04.

I specifically created exercise 05 so people know it's a separate exercise.
- It's totally focused on the platform, not so much on the architectural ideas from the modules presented
- It's for those who finish all their exercises early. But those attendees who can't finish exercise 04 in time, they won't feel bad about not finishing an exercise.
- It's quite large to 'just' include it at the end of module 04.
- People can pick it up at home, if they're really eager to experiment with the exercises.
- We could even point people not doing the workshop to this exercises, for example via support cases or LiveChat.

Module 01 & 02 could perhaps use some advanced exercises as well. But I was thinking more about doing module 01 to 04 as advanced modules and not include 100% of the code to copy-n-paste.

**Comment by adamralph** [comment]
Discussed with @dvdstelt. He will send another PR showing an alternative approach with SI and SP content in advanced exercise sections spread across the current exercises 2,3, and 4. Marking this PR as WIP while we decided between the two approaches.

**Comment by adamralph** [comment]
Closing in favour of #126

**Pull request closed without merging**